_posts/2023-04-19-12.0.0-release.md

```diff
@@ -0,0 +1,79 @@
+---
+layout: post
+title: "Apache Arrow 12.0.0 Release"
+date: "2023-04-19 00:00:00"
+author: pmc
+categories: [release]
+---
+<!--
+{% comment %}
+Licensed to the Apache Software Foundation (ASF) under one or more
+contributor license agreements.  See the NOTICE file distributed with
+this work for additional information regarding copyright ownership.
+The ASF licenses this file to you under the Apache License, Version 2.0
+(the "License"); you may not use this file except in compliance with
+the License.  You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+{% endcomment %}
+-->
+
+
+The Apache Arrow team is pleased to announce the 12.0.0 release. This covers
+over 3 months of development work and includes [**XXX resolved issues**][1]
+from [**YYY distinct contributors**][2]. See the [Install Page](https://arrow.apache.org/install/)
+to learn how to get the libraries for your platform.
+
+The release notes below are not exhaustive and only expose selected highlights
+of the release. Many other bugfixes and improvements have been made: we refer
+you to the [complete changelog][3].
+
+## Community
+
+Since the 11.0.0 release, Wang Mingming, Mustafa Akur and Ruihang Xia
+have been invited to be committers.
+Will Jones have joined the Project Management Committee (PMC).
+
+Thanks for your contributions and participation in the project!
+
+## Columnar Format Notes
+
+## Arrow Flight RPC notes
+
+## C++ notes
+
+## C# notes
+
+## Go notes
```

Comment:
@zeroshade here we are again :)

Reply:
```suggestion
## Go notes

* Run-End Encoded Arrays have been added to the Golang implementation [GH-32104](https://github.com/apache/arrow/issues/32104) [GH-32946](https://github.com/apache/arrow/issues/32946) [GH-20407](https://github.com/apache/arrow/issues/20407) [GH-32949](https://github.com/apache/arrow/issues/32949)
* The SQLite Flight SQL Example has been improved and you can now `go get` a simple SQLite Flight SQL Server mainprog using `go get github.com/apache/arrow/go/v12/arrow/flight/flightsql/example/cmd/sqlite_flightsql_server` [GH-33840](https://github.com/apache/arrow/issues/33840)
* Fixed bug causing builds to fail with the `noasm` build tag [GH-34044](https://github.com/apache/arrow/issues/34044) and added a CI test run that uses the `noasm` tag [GH-34055](https://github.com/apache/arrow/issues/34055)
* Fixed issue allowing building on riscv64-freebsd [GH-34629](https://github.com/apache/arrow/issues/34629)
* Fixed issue preventing building on 32-bit platforms [GH-35133](https://github.com/apache/arrow/issues/35133)

### Arrow

* Fixed bug in C-Data handling of `ArrowArrayStream.get_next` when handling uninitialized `ArrowArrays` [GH-33767](https://github.com/apache/arrow/issues/33767)
* _Breaking Change_ Added `Err()` method to `RecordReader` interface so that it can propagate errors [GH-33789](https://github.com/apache/arrow/issues/33789)
* Fixed potential panic in C-Data API for misusing an invalid handle [GH-33864](https://github.com/apache/arrow/issues/33864)
* A new cgo-based Allocator that does not depend on libarrow has been added to the memory package [GH-33901](https://github.com/apache/arrow/issues/33901)
* CSV Reader and Writer now support Extension type arrays [GH-34334](https://github.com/apache/arrow/issues/34334)
* Fixed bug preventing the reading of IPC streams/files with compression enabled but uncompressed buffers [GH-34385](https://github.com/apache/arrow/issues/34385)
* Added interface which can be added to an `ExtensionType` to allow Builders to be created via `NewBuilder`, enabling easy building of nested fields containing extension types [GH-34453](https://github.com/apache/arrow/issues/34453)
* Added utilities to perform Array diffing [GH-34790](https://github.com/apache/arrow/issues/34790)
* Added `SetColumn` method to `arrow.Record` [GH-34832](https://github.com/apache/arrow/issues/34832)
* Added `GetValue` method to `arrow.Metadata` [GH-34855](https://github.com/apache/arrow/issues/34855)
* Added `Pow` method to `decimal128.Num` and `decimal256.Num` [GH-34863](https://github.com/apache/arrow/issues/34863)

#### Flight
* Fixed bug in `StreamChunks` for Flight SQL to correctly propagate to the gRPC client [GH-33717](https://github.com/apache/arrow/issues/33717)
* Fixed issue that prevented compatibility with gRPC < v1.45 [GH-33734](https://github.com/apache/arrow/issues/33734)
* Added support to bind a `RecordReader` for supplying parameters to a Flight SQL Prepared statement [GH-33794](https://github.com/apache/arrow/issues/33794)
* Prepared Statement methods for FlightSQL client now allows gRPC Call Options [GH-33867](https://github.com/apache/arrow/issues/33867)
* FlightSQL Extensions have been implemented (for transactions and Substrait support) [GH-33935](https://github.com/apache/arrow/issues/33935)
* A driver compatible with `database/sql` for FlightSQL has been added [GH-34332](https://github.com/apache/arrow/issues/34332)

#### Compute

* "run_end_encode" and "run_end_decode" functions added to compute functions [GH-20408](https://github.com/apache/arrow/issues/20408)
* "unique" function added [GH-34171](https://github.com/apache/arrow/issues/34171)

### Parquet
* `pqarrow` pkg now handles DICTIONARY fields natively [GH-33466](https://github.com/apache/arrow/issues/33466)
* Fixed rare panic when writing list of 8 structs [GH-33600](https://github.com/apache/arrow/issues/33600)
* Added support for `pqarrow` pkg to write LargeString and LargeBinary types [GH-33875](https://github.com/apache/arrow/issues/33875)
* Fixed bug where `pqarrow.NewSchemaManifest` created the wrong field type for Array Object fields [GH-34101](https://github.com/apache/arrow/issues/34101)
* Added support to Parquet Writer for Extension type Arrays [GH-34330](https://github.com/apache/arrow/issues/34330)
```